runtime/doc/lua.txt

```diff
@@ -1436,6 +1436,41 @@ notify_once({msg}, {level}, {opts})                        *vim.notify_once()*
     Return: ~
         (boolean) true if message was displayed, else false
 
+on_fun({container}, {key}, {fn})                                *vim.on_fun()*
+    Sets function `container[key]` to a new (wrapper) function that, if
+    invoked, calls `fn()` before optionally calling the original ("base")
+    function.
```

Comment:
I'm surprised I couldn't find something like this in https://github.com/rxi/lume or https://github.com/luafun/luafun .

Does anyone have a util function like this in their plugins?